core/src/main/java/jenkins/util/java/JavaUtils.java

```diff
@@ -51,8 +52,23 @@ public static boolean isRunningWithJava8OrBelow() {
      * @return {@code true} if it is Java 9 or above
      */
     public static boolean isRunningWithPostJava8() {
-        String javaVersion = System.getProperty("java.specification.version");
+        String javaVersion = getCurrentRuntimeJavaVersion();
         return !javaVersion.startsWith("1.");
     }
 
+    /**
+     * Returns the JVM's current version as a {@link VersionNumber} instance.
```

Comment:
Best to document this with examples. IIUC these would be

* `1.8`
* `9`
* `10`
* `11`

?

Reply:
Yes. Basically [JEP-223](https://openjdk.java.net/jeps/223)

Reply:
@jglick fix proposed in https://github.com/jenkinsci/jenkins/pull/3842